geo/src/algorithm/affine_ops.rs

```diff
@@ -1,6 +1,59 @@
-use crate::{CoordFloat, CoordNum, Coordinate, Point};
+use crate::{CoordFloat, CoordNum, Coordinate, MapCoords, MapCoordsInPlace};
 use std::fmt;
 
+/// Apply an [`AffineTransform`] like [`scale`](AffineTransform::scale),
+/// [`skew`](AffineTransform::skew), or [`rotate`](AffineTransform::rotate) to a
+/// [`Geometry`](crate::geometry::Geometry).
+///
+/// Multiple transformations can be composed in order to be efficiently applied in a single
+/// operation. See [`AffineTransform`] for more on how to build up a transformation.
+///
+/// For some common transformations, existing traits leverage this same machinery, but might be
+/// more readable. See: [`Scale`](crate::algorithm::Scale),
+/// [`Translate`](crate::algorithm::Translate), [`Rotate`](crate::algorithm::Rotate), and
+/// [`Skew`](crate::algorithm::Skew).
+///
+/// # Examples
+/// ## Build up transforms by beginning with a constructor, then chaining mutation operations
+/// ```
+/// use geo::{AffineOps, AffineTransform};
+/// use geo::{line_string, BoundingRect, Point, LineString};
+/// use approx::assert_relative_eq;
+///
+/// let ls: LineString = line_string![
+///     (x: 0.0f64, y: 0.0f64),
+///     (x: 0.0f64, y: 10.0f64),
+/// ];
+/// let center = ls.bounding_rect().unwrap().center();
+///
+/// let transform = AffineTransform::skew(40.0, 40.0, center).rotated(45.0, center);
+///
+/// let skewed_rotated = ls.affine_transform(&transform);
+///
+/// assert_relative_eq!(skewed_rotated, line_string![
+///     (x: 0.5688687f64, y: 4.4311312),
+///     (x: -0.5688687, y: 5.5688687)
+/// ], max_relative = 1.0);
+/// ```
+pub trait AffineOps<T: CoordNum> {
+    /// Apply `transform` to mutate `self`.
+    fn affine_transform_mut(&mut self, transform: &AffineTransform<T>);
+
+    /// Apply `transform` immutably, outputting a new geometry.
+    #[must_use]
+    fn affine_transform(&self, transform: &AffineTransform<T>) -> Self;
```

Comment:
Originally I wanted to do:

mutable: `affine_transform` 
immutable: `affine_transformed`

But it just conflicts too much with other implementations. I'm going to follow up with another PR to add mutable flavors with consistent naming for Scale/Skew/Rotate/Translate.